yunwoo/programmers-2/프린터.js

```diff
@@ -0,0 +1,23 @@
+function solution(priorities, location) {
+  const requestsAndCurrentIndexs = priorities.map((el, idx) => [el, idx]);
+  let result = 0;
+
+  while (requestsAndCurrentIndexs.length) {
+    const request = requestsAndCurrentIndexs.shift();
+
+    if (requestsAndCurrentIndexs.some((el) => el[0] > request[0])) {
+      requestsAndCurrentIndexs.push(request);
+    } else {
+      result++;
+      if (request[1] === location) break;
+    }
+  }
+
+  return result;
+}
```

Comment:
```suggestion
function solution(priorities, location) {
  const requestsAndCurrentIndexs = priorities.map((el, idx) => [el, idx]);
  let result = 0;

  while (requestsAndCurrentIndexs.length) {
    const firstEl = requestsAndCurrentIndexs.shift();
    const isNonFirstElMaxPriority = requestsAndCurrentIndexs.some((el) => el[0] > firstEl[0]);

    if (isNonFirstElMaxPriority) {
      requestsAndCurrentIndexs.push(firstEl);
    } else {
      result++;
      if (firstEl[1] === location) break;
    }
  }

  return result;
}
```
- 풀이잘봤습니다 🙇‍♂️
- 조건식에 들어가는 부분도 변수로 빼서 가독성을 키워주세요!~ 
- 효율성은 좀 안좋아지긴 하는데, 아래와 같이 max를 이용한 풀이도 있습니다.
```js
function solution(priorities, location) {
  const convertedPriorities = priorities.map((el, i) => [i, el]);
  const stack = [];

  while (convertedPriorities.length) {
    const maxPriority = Math.max(...convertedPriorities.map((el) => el[1]));
    const firstEl = convertedPriorities.shift();

    if (firstEl[1] !== maxPriority) {
      convertedPriorities.push(firstEl);
    } else {
      stack.push(firstEl);
    }
  }

  return stack.findIndex((el) => el[0] === location) + 1;
}
```

Reply:
- 아하 넵 다음엔 변수로 빼보겠습니다 👍
- 참고풀이도 잘보겠습니다 감사합니다🙇‍♂️